runtime/tests/runtime_integration_tests/test_get_img_info_warmreset.rs

```diff
@@ -0,0 +1,150 @@
+
+use crate::test_set_auth_manifest::{
+    create_auth_manifest, create_auth_manifest_with_metadata, AuthManifestBuilderCfg,
+};
+
+use caliptra_auth_man_types::{
+    Addr64, AuthManifestFlags, AuthManifestImageMetadata, AuthorizationManifest, ImageMetadataFlags,
+};
+
+use caliptra_builder::ImageOptions;
+
+use caliptra_common::mailbox_api::{
+    CommandId, GetImageInfoReq, GetImageInfoResp, ImageHashSource, MailboxReq, MailboxReqHeader,
+    SetAuthManifestReq,
+};
+
+use caliptra_hw_model::{DefaultHwModel, HwModel};
+
+use caliptra_image_types::FwVerificationPqcKeyType;
+
+use zerocopy::{FromBytes, IntoBytes};
+
+const FW_ID_1: u32 = 1;
+const FW_ID_2: u32 = 2;
+
+/// Helper: issue GET_IMAGE_INFO for a given fw_id and return a *owned* response.
+fn get_image_info(model: &mut DefaultHwModel, fw_id: u32) -> GetImageInfoResp {
+    let mut get_image_info_cmd = MailboxReq::GetImageInfo(GetImageInfoReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        fw_id: fw_id.to_le_bytes(),
+    });
+
+    get_image_info_cmd.populate_chksum().unwrap();
+
+    let resp = model
+        .mailbox_execute(
+            u32::from(CommandId::GET_IMAGE_INFO),
+            get_image_info_cmd.as_bytes().unwrap(),
+        )
+        .unwrap()
+        .expect("GET_IMAGE_INFO should return a response");
+
+    // Use read_from_bytes to get an owned struct (easier to compare later).
+    GetImageInfoResp::read_from_bytes(resp.as_slice()).expect("failed to parse GetImageInfoResp")
+}
+
+fn set_auth_manifest(auth_manifest: Option<AuthorizationManifest>) -> DefaultHwModel {
+    let runtime_args = RuntimeTestArgs {
+        test_image_options: Some(ImageOptions {
+            pqc_key_type: FwVerificationPqcKeyType::LMS,
+            ..Default::default()
+        }),
+        ..Default::default()
+    };
+
+    let mut model = run_rt_test(runtime_args);
+    wait_runtime_ready(&mut model);
+
+    let auth_manifest = if let Some(auth_manifest) = auth_manifest {
+        auth_manifest
+    } else {
+        create_auth_manifest(&AuthManifestBuilderCfg {
+            manifest_flags: AuthManifestFlags::VENDOR_SIGNATURE_REQUIRED,
+            pqc_key_type: FwVerificationPqcKeyType::LMS,
+            ..Default::default()
+        })
+    };
+
+    let buf = auth_manifest.as_bytes();
+    let mut auth_manifest_slice = [0u8; SetAuthManifestReq::MAX_MAN_SIZE];
+    auth_manifest_slice[..buf.len()].copy_from_slice(buf);
+
+    let mut set_auth_manifest_cmd = MailboxReq::SetAuthManifest(SetAuthManifestReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        manifest_size: buf.len() as u32,
+        manifest: auth_manifest_slice,
+    });
+    set_auth_manifest_cmd.populate_chksum().unwrap();
+
+    model
+        .mailbox_execute(
+            u32::from(CommandId::SET_AUTH_MANIFEST),
+            set_auth_manifest_cmd.as_bytes().unwrap(),
+        )
+        .unwrap()
+        .expect("We should have received a response");
+
+    model
+}
```

Comment:
I see this already duplicated once in other tests. Use https://github.com/chipsalliance/caliptra-sw/pull/2853 ?